### PR TITLE
Add embind code size test. NFC

### PR DESCRIPTION
**Pull request by sbc100**
In order to make this new test even compile I moved the declaration of `dynCalls` from being hand coded in python to being decoratively included in library.js, which makes more sense anyway.

See #22416

### CODE REVIEW

File: tools/emscripten.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def compute_minimal_runtime_initializer_and_exports(post, exports, receiving):
   exports = [asmjs_mangle(x) for x in exports if x != building.WASM_CALL_CTORS]
 
   # Decide whether we should generate the global dynCalls dictionary for the dynCall() function?
-  if settings.DYNCALLS and '$dynCall' in settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE and len([x for x in exports if x.startswith('dynCall_')]) > 0:
+  if settings.DYNCALLS or not settings.WASM_BIGINT or '$dynCall' in settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE:
```
Comment (brendandahl) [inline tools/emscripten.py:64]: Is this change suppose to be in here?

Comment (sbc100) [reply tools/emscripten.py:64]: Yeah, without this the simple test doesn't compile due to closure warning about `dynCalls` symbol being missing.

Comment (sbc100) [reply tools/emscripten.py:64]: Actually I found a better way to express this dependency precisely in library.js where it belongs.

**Review by brendandahl — APPROVED** [review]

File: src/library.js [hunk 2]
```diff
@@ -1856,14 +1856,21 @@ addToLibrary({
   },
 
 #if DYNCALLS || !WASM_BIGINT
+#if MINIMAL_RUNTIME
+  $dynCalls: '[]',
```
Comment (kripken) [inline src/library.js:1860]: This was `{}` before, and it does seem to be used more like a map than an array - why was this changed?

Comment (sbc100) [reply src/library.js:1860]: Oops you are right... it should be {}
